src/components/button/_macro.spec.js

```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
-            const $ = cheerio.load(
-                renderComponent('button', {
-                    text: 'Example button',
-                    name: 'example',
-                    value: 'example-value',
-                }),
-            );
-
-            const results = await axe($.html());
-            expect(results).toHaveNoViolations();
+    describe('GIVEN: Params: variants', () => {
+        describe('WHEN: variants are present', () => {
+            it('THEN: renders provided variant style classes', () => {
+                const $ = cheerio.load(
+                    renderComponent('button', {
+                        variants: ['variant-a', 'variant-b'],
+                    }),
+                );
+
+                expect($('.ons-btn').hasClass('ons-btn--variant-a')).toBe(true);
+                expect($('.ons-btn').hasClass('ons-btn--variant-b')).toBe(true);
+            });
         });
+        describe('WHEN: variants contains download', () => {
+            it('THEN: renders provided variant style class', () => {
+                const $ = cheerio.load(
+                    renderComponent('button', {
+                        url: 'http://example.com',
+                        variants: 'download',
+                    }),
+                );
+                expect($('.ons-btn').hasClass('ons-btn--download')).toBe(true);
+            });
+            it('THEN: renders download icon', () => {
+                const faker = templateFaker();
+                const iconsSpy = faker.spy('icon');
 
-        it('is an `button` element', () => {
-            const $ = cheerio.load(renderComponent('button'));
+                faker.renderComponent('button', {
+                    url: 'http://example.com',
+                    variants: 'download',
+                });
 
-            expect($('button').length).toBe(1);
+                expect(iconsSpy.occurrences[0].iconType).toBe('download');
+            });
+            it('THEN: renders the download attribute when variants contains download', () => {
+                const $ = cheerio.load(
+                    renderComponent('button', {
+                        variants: 'download',
+                    }),
+                );
+
+                expect($('.ons-btn').attr('download')).toBeDefined();
+            });
         });
+        describe('WHEN: variants contains loader', () => {
+            it('THEN: renders provided variant style classes', () => {
+                const $ = cheerio.load(
+                    renderComponent('button', {
+                        variants: 'loader',
+                    }),
+                );
+
+                expect($('.ons-btn').hasClass('ons-btn--loader')).toBe(true);
+                expect($('.ons-btn').hasClass('ons-js-loader')).toBe(true);
+                expect($('.ons-btn').hasClass('ons-js-submit-btn')).toBe(true);
+            });
+            it('THEN: loader icon when variants contains loader', () => {
```

Comment:
```suggestion
            it('THEN: the button has the loader icon', () => {
```
Already in the above when